### PR TITLE
[Fix] where errors on zero dim cpu tensor inputs

### PR DESCRIPTION
**Pull request by tongxin**
### PR Category

Operator

### Type of Change

Bug Fix and New Feature

### Description

Now `where` is able to handle cross device inputs with ndim=0.

Error case,

```
cond = torch.randint(0, 2, (10, ), dtype=torch.bool, device='cuda')

torch.where(cond, torch.tensor(0), torch.tensor(1))
```

### Progress

- [x] Change is properly reviewed (1 reviewer required, 2 recommended).
- [x] Change is responded to an issue.
- [x] Change is fully covered by a UT.

### CODE REVIEW

**Review by Bowen12992 — APPROVED** [review]
LGTM
